bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,172 @@
+from coalib.parsing.StringProcessing.Core import (search_for,
+                                                  search_in_between,
+                                                  unescaped_search_in_between)
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.HiddenResult import HiddenResult
+
+
+class AnnotationBear(LocalBear):
+
+    def run(self, filename, file: str, language: str, language_family: str):
+        """
+        Finds out all the positions of comments and strings in a file.
+        The Bear searches for valid comments and strings
+        and yields them as HiddenResults with the appropriate ranges.
+        Invalid comments or strings are:
+        1. Comments within comments,
+        2. Comments within strings and
+        3. Strings within comments.
+        """
+        file = ''.join(file)
+        annot_dict = {}
+        annot_dict['comment'] = {}
+        annot_dict['string'] = {}
+        annot_dict['string']['multiline'] = {}
+        annot_dict['string']['singleline'] = {}
+        annot_dict['comment']["singleline"] = {}
+        annot_dict['comment']["multiline"] = {}
+        # Tuple of tuples containing start and end of annotations
+        match_pos = ()
+        lang_dict = LanguageDefinition(language, language_family)
+
+        annot_dict['comment']["singleline"].update(
+                    lang_dict["comment_delimiter"])
+
+        annot_dict['comment']["multiline"].update(
+                    lang_dict["multiline_comment_delimiters"])
+
+        annot_dict['string']['singleline'].update(
+                                            lang_dict["string_delimiters"])
+        annot_dict['string']['multiline'].update(
+                                       lang_dict["multiline_string_delimiters"])
+        multi_str = single_str = multi_comm = ()
+        search_dict = annot_dict['string']['multiline']
+        multi_str = find_string_comment(file, search_dict, True)
+        search_dict = annot_dict['string']['singleline']
```

Comment:
The variable `search_dict` seems completely unnecessary and is confusing because it keeps changing and is only used once before it gets changed.